sklearn/cluster/_kmeans.py

```diff
@@ -426,6 +426,12 @@ def kmeans_single(self, X, sample_weight, centers_init):
 
     def prepare_prediction(self, X, sample_weight):
         X = self.estimator._check_test_data(X)
+
+        # The deprecation check and raising the warning happens in the estimator,
+        # so we just have to handle the case of a string being passed in.
+        if sample_weight == "deprecated":
+            sample_weight = None
+
```

Comment:
I fixed the other part, but this part might be worth keeping.

Reply:
Actually I did it too but in a different manner.

Reply:
You dealt with the "conversion" from `"deprecated"` to the default value (`None`) in the estimator. This means that the engine never sees the special string "deprecated". I think I prefer this, even though it means the engine authors need to look at the estimator to find out about deprecations. Having to look at the estimator to learn about deprecations seems easier than getting every engine author to think about how to handle `"deprecated"`, which means overall the chances of an engine breaking because of a deprecation are lower.